Workbooks/WorkItems/Azure DevOps/Azure DevOps.workbook

```diff
@@ -0,0 +1,221 @@
nion query\r\n`<<telemetrytable>>` - requests, traces, etc.\r\n`bodyOfWit` - The string you want to be added to the work item or issue.\r\n`nameForTitle` - Used to generate the correct title for a specific telemetry type. For example, avalabilityResults may not have an operation_Name defined.\r\n`workItemKey` - This is always 1 and is used to automatically generate the correct list of work item types (if you want multiple) for Azure DevOps. See below in the union section.\r\n\r\n```\r\nlet <<anyname>> = <<telemetrytable>> | where timestamp between(starttime..endtime) | where itemId == itemIdFromWorkbook\r\n| extend bodyOfWit = strcat(returnline(\"Operation Name\", operation_Name), returnline(\"Timestamp\", timestamp), returnline(\"Duration (ms)\", duration), returnline(\"Country or region\", client_CountryOrRegion), returnline(\"State or province\", client_StateOrProvince), returnline(\"City\", client_City), returnline(\"Link\", urlToDetails))\r\n| extend nameForTitle = name\r\n| extend workItemKey = 1;\r\n\r\n```\r\n### workItemTypes datatable \r\n\r\nIn order to give you the option to create a Bug for one type of telemetry and a task for another type we generate a data table that has a list of workItemTypes with a hard coded id of 1. We join with the hardcoded key so we can repeat the rows for each work item type and generate the correct URL with the correct parameters. You can see this in the query and it looks like this.\r\n\r\n```\r\nlet workItemTypes = datatable ( workItemType: string, workItemKey: long )['Bug', 1, 'Task', 1];\r\n```\r\n\r\nIf you want to add another work item type or only have one you can modify the data table to reflect any built in or custom type.\r\n\r\n### Union of direct queries\r\n\r\nAfter the direct queries are defined we combine them into one unified query to generate the correct table. For Azure DevOps we do one special thing here and that is join the uninoned query with a custom datatable called workItemType. This simply repeat the row so we can create multiple work item types if desired.\r\n\r\n## Troubleshooting\r\n\r\n### 404 Page Not Found\r\n\r\nThis indicates that either the organization name is incorrect or the project is incorrect. Check the URL you entered into the wizard and try again.\r\n\r\n### 403 Unauthorized\r\n\r\nThis means that you do not have access to the organization or the project. Please validate you have the correct URL and if so contact your administrator to get access.\r\n\r\n### Work Item Data is truncated\r\n\r\nThere are two reasons for this and you only have a few options to correct them.\r\n\r\n**Data Length** - For most web servers there is a max URL length. If your data is particularly long or if there is a URL length restriction you may not see the end of your data\r\n**URL Encoding** - By default we URL encode the values going into the `returnline` function. But if you alter this or if you are doing something custom you should check to see if using url_encode() fixes the issue"
```

Comment:
some typos in  'document whic', 'generats', 'querystirng'... etc. Should do a final scan here to double check

Reply:
updated.